src/monitoring_service/cli.py

```diff
@@ -154,9 +154,15 @@ def main(
     )
 
     ms.start()
-    gevent.spawn(check_gas, web3, private_key)
+    check_gas_task = gevent.spawn(check_gas, web3, private_key)
+
+    # this blocks until the service is stopped
     ms.start(gevent.sleep)
 
+    # stop the remaining task
+    check_gas_task.kill()
```

Comment:
Do we kill the Greenlet right after starting the MS? In that case, it would only be executed once (because the first execution is scheduled before the kill).